One of the tests need patching:

  - Because we're doing the build in a symlink farm, one of the diff tests
    thinks that we're changing the type of the files from symlink to file,
    so the diffs come out wrong.  Rather than cleverly mucking around with
    the working directory vs the repo directory, we just cat the files as
    real files into place.

Unclear whether this patch should be sent upstream.

--- git-2.50.1/t/Makefile.orig	2025-07-09 11:25:02.515332981 +0200
+++ git-2.50.1/t/Makefile	2025-07-09 11:29:12.228513779 +0200
@@ -62,6 +62,9 @@
 test: pre-clean check-meson $(TEST_LINT)
 	$(CHAINLINTSUPPRESS) $(MAKE) aggregate-results-and-cleanup
 
+test-noclean: pre-clean $(TEST_LINT)
+	$(MAKE) T
+
 ifneq ($(PERL_PATH),)
 test: check-chainlint
 prove: check-chainlint
@@ -161,6 +164,8 @@
 test-lint-shell-syntax:
 	@'$(PERL_PATH_SQ)' check-non-portable-shell.pl $(T) $(THELPERS) $(TPERF)
 
+T: $(T)
+
 test-lint-filenames:
 	@# We do *not* pass a glob to ls-files but use grep instead, to catch
 	@# non-ASCII characters (which are quoted within double-quotes)
--- git-2.48.1/t/t4018-diff-funcname.sh.orig	2025-02-05 17:29:20.578919102 +0100
+++ git-2.48.1/t/t4018-diff-funcname.sh	2025-02-05 17:36:12.229457645 +0100
@@ -99,8 +99,11 @@
 
 	# add all test files to the index
 	(
-		cd "$TEST_DIRECTORY"/t4018 &&
-		git --git-dir="$TRASH_DIRECTORY/.git" add .
+		( cd "$TEST_DIRECTORY"/t4018 && printf "%s\n" * ) | while read f
+				do
+					cat "$TEST_DIRECTORY"/t4018/$f > $f
+					git add -v $f
+				done	
 	) &&
 
 	# place modified files in the worktree
